stl/inc/algorithm

```diff
@@ -7520,6 +8070,268 @@ void stable_sort(_ExPo&& _Exec, _BidIt _First, _BidIt _Last) noexcept /* termina
+        static void _Buffered_merge_sort_common(const _It _First, const _It _Last, const iter_difference_t<_It> _Count,
+            iter_value_t<_It>* const _Temp_ptr, _Pr _Pred, _Pj _Proj) {
+            // sort using temp buffer for merges
+            // pre: _Count <= capacity of buffer at _Temp_ptr; also allows safe narrowing to ptrdiff_t
+            _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+            _STL_INTERNAL_CHECK(_Last - _First == _Count);
+
+            _Insertion_sort_isort_max_chunks(_First, _Last, _Count, _Pred, _Proj);
+            // merge adjacent pairs of chunks to and from temp buffer
+            if (_Count <= _Isort_max<_It>) {
+                return;
+            }
+
+            // do the first merge, constructing elements in the temporary buffer
+            _Uninitialized_chunked_merge_common(_First, _Last, _Temp_ptr, _Count, _Pred, _Proj);
+            _Uninitialized_backout<iter_value_t<_It>*> _Backout{_Temp_ptr, _Temp_ptr + _Count};
+            iter_difference_t<_It> _Chunk_size = _Isort_max<_It>;
+            for (;;) {
+                // unconditionally merge elements back into the source buffer
+                _Chunk_size <<= 1;
+                _Chunked_merge_common(_Temp_ptr, _Temp_ptr + _Count, _First, _Chunk_size, _Count, _Pred, _Proj);
+                _Chunk_size <<= 1;
+                if (_Count <= _Chunk_size) { // if the input would be a single chunk, it's already sorted and we're done
+                    return;
+                }
+
+                // more merges necessary; merge to temporary buffer
+                _Chunked_merge_common(_First, _Last, _Temp_ptr, _Chunk_size, _Count, _Pred, _Proj);
+            }
+        }
+
+        template <class _It, class _Pr, class _Pj>
+        static void _Insertion_sort_isort_max_chunks(
+            _It _First, _It _Last, iter_difference_t<_It> _Count, _Pr _Pred, _Pj _Proj) {
+            // insertion sort every chunk of distance _Isort_max<_It> in [_First, _Last)
+            _STL_INTERNAL_STATIC_ASSERT(random_access_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+            _STL_INTERNAL_CHECK(_RANGES distance(_First, _Last) == _Count);
+
+            for (; _Isort_max<_It> < _Count; _Count -= _Isort_max<_It>) { // sort chunks
+                _First = _RANGES _Insertion_sort_common(_First, _First + _Isort_max<_It>, _Pred, _Proj);
+            }
+
+            // sort partial last chunk
+            _RANGES _Insertion_sort_common(_STD move(_First), _STD move(_Last), _Pred, _Proj);
+        }
+
+        template <class _It, class _Pr, class _Pj>
+        static void _Uninitialized_chunked_merge_common(_It _First, const _It _Last, iter_value_t<_It>* const _Dest,
+            iter_difference_t<_It> _Count, _Pr _Pred, _Pj _Proj) {
+            // move to uninitialized merging adjacent chunks of distance _Chunk

```

Comment:
Ah, I see. `_Chunk` is the name of a parameter for `std::_Uninitialized_chunked_merge_unchecked`. It has only one callsite, and the value passed is always `iter_difference_t<_It>{_ISORT_MAX}` (which in the `std::ranges` flavor is spelled `_Isort_max<_It>`). @miscco constant-propagated the value through this function when he duplicated it from `std`.

It would be nice to rename `_Chunk2` to `_Chunk` (now that it's the only `_ChunkX`) and change the `std` algorithm similarly to keep them both consistent, which I think I'll do.